src/hotspot/share/utilities/growableArray.hpp

```diff
@@ -720,15 +723,31 @@ class GrowableArray : public GrowableArrayWithAllocator<E, GrowableArray<E> > {
   }
 
 public:
-  GrowableArray(int initial_capacity = 2, MEMFLAGS memflags = mtNone) :
+  GrowableArray(int initial_capacity = 2) :
+      GrowableArrayWithAllocator<E, GrowableArray<E> >(
+          allocate(initial_capacity),
+          initial_capacity),
+      _metadata() {
+    init_checks();
+  }
+
+  GrowableArray(MEMFLAGS memflags, int initial_capacity = 2) :
```

Comment:
I feel it's too dangerous to have two constructors with convertible parameters. 
eg. GrowableArray(int initial_capacity)  and   GrowableArray(MEMFLAGS memflags, int initial_capacity = 2) are very close. what if someone has code like this:
```
short t = 2;
GrowableArray(2); // it's very vague to me which ctor will be selected.
```

I think it's a good idea to use 'explicit constructor' for one of them. in case implicit conversation takes place unconsciously.

Reply:
MEMFLAGS are defined as a class enum, so integers will not be automatically converted to MEMFLAGS. I'll add explicit to the constructors.